quipucords/cli/clicommand.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python
+#
+# Copyright (c) 2017 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public License,
+# version 3 (GPLv3). There is NO WARRANTY for this software, express or
+# implied, including the implied warranties of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv3
+# along with this software; if not, see
+# https://www.gnu.org/licenses/gpl-3.0.txt.
+#
+""" Base CLI Command Class """
+
+from __future__ import print_function
+
+
+# pylint: disable=too-few-public-methods
+class CliCommand(object):
+
+    """ Base class for all sub-commands. """
+    def __init__(self, subcommand, action):
+        self.subcommand = subcommand
+        self.action = action
+        self.args = None
+
+    def _validate_args(self):
```

Comment:
Nit: can we please remove the leading `_` from this and `do_command`? Those always bothered me in Rho.

Reply:
is that not convention for denoting a method as private/internal? Meaning those two methods should not be called directly.

Reply:
Yes, that's the convention. Maybe it bothers me because it's meant to be overridden, I'm not sure. Anyway, LGTM.